### PR TITLE
Add manual implementation of EntryCompletion::set_match_func

### LINKED ISSUE #685: EntryCompletion missing set_match_func
[GTK docs](https://developer.gnome.org/gtk3/stable/GtkEntryCompletion.html#gtk-entry-completion-set-match-func)
Might be tricky to Rustify since you have to pass it a function pointer, but I need access to this to customize the completion behavior.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #685.

Two things about this implementation:
1. It doesn't take into account the destroy notify
2. It'll certainly be removed "soon" when we'll generate callbacks as well.

cc @EPashkin @sdroege

### CODE REVIEW

File: src/entry_completion.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pub trait EntryCompletionExtManual {
+    fn set_match_func<F: Fn(&Self, &str, &TreeIter) -> bool + 'static>(&self, f: F);
+}
+
+impl<O: IsA<EntryCompletion>> EntryCompletionExtManual for O {
+    fn set_match_func<F: Fn(&Self, &str, &TreeIter) -> bool + 'static>(&self, f: F) {
+        unsafe {
+            let f: Box_<Box_<Fn(&Self, &str, &TreeIter) -> bool + 'static>> = Box_::new(Box_::new(f));
+            let callback = transmute(set_match_func_trampoline::<Self> as usize);
+            ffi::gtk_entry_completion_set_match_func(self.to_glib_none().0,
```
Comment (sdroege) [inline src/entry_completion.rs]: Can be FnMut and you must use the destroyed notify to not leak.

Just do the same I did for the function passed to bind_model() of GtkListBox. Exactly the same pattern here

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by EPashkin** [comment]
```
error[E0596]: cannot borrow immutable borrowed content `**f` as mutable
  --> src/entry_completion.rs:43:5
   |
43 |     f(&EntryCompletion::from_glib_borrow(this).downcast_unchecked(),
   |     ^ cannot borrow as mutable
error: aborting due to previous error
```
also we need change clippy test as seems it included in nightly by default
```
$ cargo install clippy || touch clippy_failed
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading clippy v0.0.212
  Installing clippy v0.0.212
error: binary `cargo-clippy` already exists in destination
Add --force to overwrite
```

**Comment by GuillaumeGomez** [comment]
@EPashkin Where did you get this log? O.o CI just started

**Comment by EPashkin** [comment]
@GuillaumeGomez It may was before your second update.

**Comment by GuillaumeGomez** [comment]
Ah maybe! I was trying a few things around.

**Comment by EPashkin** [comment]
@GuillaumeGomez Please change clippy check to `  - if [ "$TRAVIS_RUST_VERSION" == "nightly" ] && [ "$GTK" == "3.4" ]; then
` too.

**Comment by GuillaumeGomez** [comment]
Yes sir!

**Comment by GuillaumeGomez** [comment]
@EPashkin Is it what you had in mind?

**Comment by GuillaumeGomez** [comment]
`clippy` was supposed to be provided with rust at some point but I didn't think it already started!

**Comment by EPashkin** [comment]
Strange
```
0.06s$ if [ "$TRAVIS_RUST_VERSION" == "nightly" ] && [ "$GTK" == "3.4" ]; then cargo clippy; fi
error: toolchain 'nightly-x86_64-unknown-linux-gnu' does not have the binary `cargo-clippy`
```

**Comment by EPashkin** [comment]
Previous version is more strange:
```
 if [ "$TRAVIS_RUST_VERSION" == "nightly" ] && [ "$GTK" == "3.4" ]; then cargo install clippy; cargo clippy; fi
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading clippy v0.0.212
  Installing clippy v0.0.212
error: binary `cargo-clippy` already exists in destination
Add --force to overwrite
error: toolchain 'nightly-x86_64-unknown-linux-gnu' does not have the binary `cargo-clippy`
```

**Comment by GuillaumeGomez** [comment]
Don't you just love dark magic? :D

**Comment by EPashkin** [comment]
Can you try add `rustup component add clippy-preview` before 'cargo clippy`?

**Comment by GuillaumeGomez** [comment]
Sure!

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks, Clippy passes, code looks good for me but IMHO better wait for @sdroege confirmation.

**Comment by GuillaumeGomez** [comment]
Fine by me. Thanks for the help on travis! :)

**Pull request merged**